src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -857,6 +857,25 @@ public IActionResult BuildTransaction([FromBody] BuildTransactionRequest request
                     });
                 }
 
+                // If specified, get the change address, which must already exist in the wallet.
+                HdAddress changeAddress = null;
+                if (!string.IsNullOrWhiteSpace(request.ChangeAddress))
```

Comment:
Out of curiosity why not put this logic in `handler.FindChangeAddress` instead of the controller then it can be reused.

Reply:
Few reasons,
- This is currently the only place that uses the change address functionality
- Requiring the change address to be in the wallet is specific to this circumstance
- Can directly return the error responses